presto-hive/src/test/java/io/prestosql/plugin/hive/TestShowStats.java

```diff
@@ -212,62 +256,145 @@ public void testShowStatsForNonExistingColumnFails()
+                        "   ('nationkey', null, 5, 0, null, 1, 24), " +
+                        "   ('name', 109, 5, 0, null, null, null), " +
+                        "   ('name2', 109, 5, 0, null, null, null), " +
+                        "   (null, null, null, null, 15, null, null))");
     }
 
     @Test
-    public void testShowStatsForNonIdentifierColumnFails()
+    public void testShowStatsForNonIdentifierColumn()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 1, 0, null, null, null), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsForColumnExpressionFails()
+    public void testShowStatsForColumnExpression()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 5, 0, null, 1, 25), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsWithGroupByFails()
+    public void testShowStatsWithGroupBy()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT avg(totalprice) FROM orders GROUP BY orderkey)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT avg(totalprice) FROM orders GROUP BY orderkey)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
     }
 
     @Test
-    public void testShowStatsWithHavingFails()
+    public void testShowStatsWithHaving()
     {
-        assertQueryFails(
+        assertQuery(
                 "SHOW STATS FOR (SELECT count(nationkey) FROM nation_partitioned GROUP BY regionkey HAVING regionkey > 0)",
-                ".*Only table columns names are supported in SHOW STATS SELECT clause");
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
     }
 
     @Test
-    public void testShowStatsWithSelectDistinctFails()
+    public void testShowStatsWithSelectDistinct()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT DISTINCT * FROM orders)", ".*DISTINCT is not supported by SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT DISTINCT * FROM orders)",
```

Comment:
Add a test with
```
SELECT DISTINCT regionkey FROM nation
```

since NDV(regionkey) is known, the row count should be known.
I think it won't work yet (the same way as the GROUP BY case above), so 
this will be a todo